assets/js/modules/search-console/components/module/ModulePopularKeywordsWidget/index.js

```diff
@@ -0,0 +1,160 @@
+
+const { useSelect } = Data;
+
+function ModulePopularKeywordsWidget( { Widget, WidgetReportZero, WidgetReportError } ) {
+	const {
+		data,
+		isLoading,
+		error,
+		serviceURL,
+	} = useSelect( ( select ) => {
+		const store = select( STORE_NAME );
+
+		const url = select( CORE_SITE ).getCurrentEntityURL();
+		const dateRangeDates = select( CORE_USER ).getDateRangeDates( { offsetDays: DATE_RANGE_OFFSET } );
+		const { startDate, endDate } = dateRangeDates;
+
+		const reportArgs = {
+			startDate,
+			endDate,
+			url: url || undefined,
+			dimensions: 'query',
+			limit: 10,
+		};
+
+		return {
+			data: store.getReport( reportArgs ),
+			isLoading: ! store.hasFinishedResolution( 'getReport', [ reportArgs ] ),
+			error: store.getErrorForSelector( 'getReport', [ reportArgs ] ),
+			serviceURL: store.getServiceReportURL( {
+				...generateDateRangeArgs( dateRangeDates ),
+				page: url ? `!${ url }` : undefined,
+			} ),
+		};
+	} );
+
+	if ( error ) {
+		return <WidgetReportError moduleSlug="search-console" error={ error } />;
+	}
+
+	if ( isZeroReport( data ) ) {
+		return <WidgetReportZero moduleSlug="search-console" />;
+	}
+
+	return (
+		<Widget
+			noPadding
+			Header={ Header }
+			Footer={ () => (
+				<SourceLink
+					className="googlesitekit-data-block__source"
+					name={ _x( 'Search Console', 'Service name', 'google-site-kit' ) }
+					href={ serviceURL }
+					external
+				/>
+			) }
+		>
+			{
+
+				isLoading ? <PreviewTable padding />
+					: (
+						<TableOverflowContainer>
+							<ReportTable rows={ data } columns={ tableColumns } />
+						</TableOverflowContainer>
+					)
+			}
+		</Widget>
+	);
+}
+
+const tableColumns = [
+	{
+		title: __( 'Keyword', 'google-site-kit' ),
+		description: __( 'Most searched for keywords related to your content', 'google-site-kit' ),
+		primary: true,
+		field: 'keys.0',
+		Component: ( { fieldValue } ) => {
+			const searchAnalyticsURL = useSelect( ( select ) => {
+				const { startDate, endDate } = select( CORE_USER ).getDateRangeDates( { offsetDays: DATE_RANGE_OFFSET } );
+				const url = select( CORE_SITE ).getCurrentEntityURL();
+				return select( MODULES_SEARCH_CONSOLE ).getServiceReportURL( {
+					...generateDateRangeArgs( { startDate, endDate } ),
+					query: `!${ fieldValue }`,
+					page: url ? `!${ url }` : undefined,
+				} );
+			} );
+
+			return (
+				<Link
+					href={ searchAnalyticsURL }
+					external
+					inherit
+				>
+					{ fieldValue }
+				</Link>
+			);
+		},
+	},
+	{
+		title: __( 'Clicks', 'google-site-kit' ),
+		description: __( 'Number of times users clicked on your content in search results', 'google-site-kit' ),
+		Component: ( { row } ) => numFmt( row.clicks, { style: 'decimal' } ),
+	},
+	{
+		title: __( 'Impressions', 'google-site-kit' ),
+		description: __( 'Counted each time your content appears in search results', 'google-site-kit' ),
+		Component: ( { row } ) => numFmt( row.impressions, { style: 'decimal' } ),
+	},
+];
```

Comment:
Although this was done in a similar way initially in related widgets, this entire array needs to be moved into the component; otherwise the strings might not get translated.